cpp/src/arrow/array/builder_base.h

```diff
@@ -98,6 +98,9 @@ class ARROW_EXPORT ArrayBuilder {
   virtual Status AppendNull() = 0;
   virtual Status AppendNulls(int64_t length) = 0;
 
+  virtual Status AppendEmpty() { return AppendNull(); }
```

Comment:
Well, we'd like AppendEmpty to precisely append a (arbitrary but defined) non-null value, so this is not correct.

Reply:
done